### PR TITLE
mbbsd: notify user after resetting password in user configuration

### PR DESCRIPTION
**Pull request by chhsiao1981**
1. move AccountRecovery::NotifyUser to passwdnotify::PasswdNotify::NotifyUser.
2. notify user after resetting password in user configuration (in addition to AccountRecovery).
3. add ip in the subject add add how-to-reset-passwd link in the template.

### CODE REVIEW

File: include/passwd_notify.hpp [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <string>
+
+namespace passwdnotify {
+
+class PasswdNotify {
```
Comment (wens) [inline include/passwd_notify.hpp]: Why would you even put in a class if there's no state...

Comment (chhsiao1981) [reply include/passwd_notify.hpp]: It's my personal preference in C++/Java (to easily know which functions are public functions).

I'll remove the class statement.

Comment (wens) [reply include/passwd_notify.hpp]: If it's not static, it's globally visible...

**Comment by wens** [comment]
Please rebase and cleanup your changes...

File: mbbsd/passwd_notify.cc [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+void NotifyUser(const std::string &userid,
+                              const std::string &fromhost,
+                              const std::string &email) {
+  std::string subject;
+  subject.append(" " BBSNAME " - ");
+  subject.append(userid);
+  subject.append("(");
+  subject.append(fromhost);
+  subject.append("), �z���K�X�w����");
```
Comment (chhsiao1981) [inline mbbsd/passwd_notify.cc:20]: This is due to github encoding issue. The original file should be ok.

**Comment by wens** [comment]
Ended up reimplementing it in C to avoid pulling in C++ dependencies just for one function.

**Pull request closed without merging**